logic/strategy/src/rel_auto_funding.rs

```diff
@@ -0,0 +1,166 @@
+    if cfg.unrealized_balance_ratio_threshold <= cfg.acceptable_unrealized_balance_ratio {
+        return Err(ValidationError::new(
+            "unrealized_balance_ratio_threshold <= acceptable_unrealized_balance_ratio",
+        ));
+    }
+
+    if cfg.max_channel_stake == BalanceType::HOPR.zero() {
+        return Err(ValidationError::new("max_channel_stake: 0"));
+    }
+
+    Ok(())
+}
+
+#[serde_as]
+#[derive(Debug, Clone, Copy, PartialEq, smart_default::SmartDefault, Validate, Serialize, Deserialize)]
+#[validate(schema(function = "validate_rel_auto_funding_cfg"))]
+pub struct RelativeAutoFundingStrategyConfig {
+    #[validate(range(min = 0.0, max = 1.0))]
+    #[serde(default = "default_unrealized_balance_threshold")]
+    #[default(default_unrealized_balance_threshold())]
+    pub unrealized_balance_ratio_threshold: f64,
+
+    #[validate(range(min = 0.0, max = 1.0))]
+    #[serde(default = "default_acceptable_unrealized_balance_ratio")]
+    #[default(default_acceptable_unrealized_balance_ratio())]
+    pub acceptable_unrealized_balance_ratio: f64,
+
+    #[serde(default = "default_max_stake")]
+    #[serde_as(as = "DisplayFromStr")]
+    #[default(default_max_stake())]
+    pub max_channel_stake: Balance,
+}
+
+pub struct RelativeAutoFundingStrategy<A, Db> {
+    chain_actions: A,
+    db: Db,
+    cfg: RelativeAutoFundingStrategyConfig,
+}
+
+impl<A, Db> Display for RelativeAutoFundingStrategy<A, Db> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", Strategy::RelativeAutoFunding(self.cfg))
+    }
+}
+
+impl<A, Db> RelativeAutoFundingStrategy<A, Db>
+where
+    A: ChannelActions,
+    Db: HoprDbChannelOperations + HoprDbTicketOperations + Clone + Send + Sync,
+{
+    pub fn new(cfg: RelativeAutoFundingStrategyConfig, db: Db, chain_actions: A) -> Self {
+        Self { chain_actions, db, cfg }
+    }
+}
+
+#[async_trait]
+impl<A, Db> SingularStrategy for RelativeAutoFundingStrategy<A, Db>
+where
+    A: ChannelActions + Sync,
+    Db: HoprDbChannelOperations + HoprDbTicketOperations + Clone + Send + Sync,
+{
+    async fn on_tick(&self) -> crate::errors::Result<()> {
+        let cfg = self.cfg;
+        let channels_to_fund = futures::stream::iter(
+            self.db
+                .get_outgoing_channels(None)
+                .await
+                .map_err(hopr_db_sql::api::errors::DbError::from)?,
+        )
+        .filter_map(|channel| {
+            let db = self.db.clone();
+            async move {
+                match db.get_unrealized_balance(&channel).await {
+                    Ok(unrealized_balance) => {
+                        let diminished_balance = channel
+                            .balance
+                            .mul_f64(cfg.unrealized_balance_ratio_threshold)
+                            .expect("impossible: unrealized_balance_ratio_threshold not in [0,1]");
```

Comment:
Would it be reasonable to strive to avoid expect in the code even if it is unlikely, by e.g. replacing it with:
```suggestion
                            .min(1.0f64)
                            .max(0.f640);
```

Reply:
There is a proper validation.

Reply:
@coderabbitai please create an issue to replace probability values in the code with a custom `Probability` types that wraps the f64 implementation, can fail on construction, but guarantees the probability range to avoid `expect` and `unwrap` code blocks.